src/AppEndToEndWithoutModuleSmartcards.test.tsx

```diff
@@ -0,0 +1,70 @@
+import React from 'react'
+import { fireEvent, render } from 'react-testing-library'
+import fetchMock from 'fetch-mock'
+
+import waitForExpect from 'wait-for-expect'
+
+import App from './App'
+
+beforeEach(() => {
+  window.localStorage.clear()
+  window.location.href = '/'
+})
+
+async function sleep(milliseconds: number) {
+  return new Promise(resolve => {
+    window.setTimeout(resolve, milliseconds)
+  })
+}
+
+it(`quick end-to-end flow`, async () => {
+  // this is what happens in demo mode
+  fetchMock.get('/card/read', 500)
+
+  const eventListenerCallbacksDictionary: any = {} // eslint-disable-line @typescript-eslint/no-explicit-any
+  window.addEventListener = jest.fn((event, cb) => {
+    eventListenerCallbacksDictionary[event] = cb
+  })
+  window.print = jest.fn(() => {
+    eventListenerCallbacksDictionary.afterprint()
+  })
+
+  const { getByText, getByTestId, queryByText } = render(<App />)
+
+  fireEvent.click(getByText('Load Sample Election File'))
+
+  // wait long enough to get the /card/read to fail and flip the demo bit
+  await sleep(250)
+
+  getByText('Scan Your Activation Code')
+  fireEvent.click(getByTestId('qrContainer'))
```

Comment:
How do you image we should get past the activation screen once we replace QR Code on the activation page with some type of card-reader graphic. :thinking:

Reply:
great q.

I think we'll want to identify a demo mode, probably in the `election.json`, where you can just tap the smart card icon. But not in a real election of course.